src/run_program.rs

```diff
@@ -1330,53 +1330,75 @@ fn test_run_program() {
 #[cfg(test)]
 use rstest::rstest;
 
+// the test cases for this test consists of:
+// prg: the program to run inside the softfork guard
+// cost: the expected cost of the program (the test adds the apply-operator)
+// enabled: the softfork extension number that enables operator in prg
+// hard_fork_flag: the flag that enables the program to be run outside the guard
+// err: the expected error message, empty string means OK
+// The test programs are carefully crafted such that they fail with "clvm raise"
+// when run in consensus mode and the operators are unknown. e.g. (coinid ...)
+// returns NIL in that case, which compares not equal to the coin ID, which
+// raises the exception.
```

Comment:
this comment explain the tweaks to some test cases, namely: `case::coinid` and `case::modpow`.